src/auxiliary/mpi_arrays.jl

```diff
@@ -0,0 +1,194 @@
+# LoopVectorization.jl etc.
+# See https://juliaarrays.github.io/ArrayInterface.jl/stable/
+ArrayInterface.parent_type(::Type{TrixiMPIArray{T, N, Parent}}) where {T, N, Parent} = Parent
+
+
+# TODO: MPI. For now, we do not implement specializations of LinearAlgebra
+#            functions such as `norm` or `dot`. We might revisit this again
+#            in the future.
+
+
+# `mapreduce` functionality from Base using global reductions via MPI communication
+function Base.mapreduce(f::F, op::Op, u::TrixiMPIArray; kwargs...) where {F, Op}
+  local_value = mapreduce(f, op, parent(u); kwargs...)
+  return MPI.Allreduce(local_value, op, mpi_comm(u))
+end
+
+
+# TODO: MPI. Default settings of OrdinaryDiffEq etc.
+# Interesting options could be
+# - ODE_DEFAULT_UNSTABLE_CHECK
+# - ODE_DEFAULT_ISOUTOFDOMAIN
+# - ODE_DEFAULT_NORM
+# See https://github.com/SciML/DiffEqBase.jl/blob/master/src/common_defaults.jl
+
+# Problems and inconsistencies with TrixiMPIArrays
+# A basic question is how to handle `length`. We want `TrixiMPIArray`s to behave
+# like regular `Array`s in most code, e.g., when looping over an array (which
+# should use `eachindex`). At the same time, we want to be able to use adaptive
+# time stepping using error estimates in OrdinaryDiffEq.jl. There, the default
+# norm `ODE_DEFAULT_NORM` is the one described in the book of Hairer & Wanner,
+# i.e., it includes a division by the global `length` of the array. We could
+# specialize `ODE_DEFAULT_NORM` accordingly, but that requires depending on
+# DiffEqBase.jl (instead of SciMLBase.jl). Alternatively, we could implement
+# this via Requires.jl, but that will prevent precompilation and maybe trigger
+# invalidations. Alternatively, could implement our own norm and pass that as
+# `internalnorm`. Here, we decide to use the least intrusive approach for now
+# and specialize `length` to return a global length while making sure that all
+# local behavior is still working as expected (if using `eachindex` instead of
+# `1:length` etc.). This means that `eachindex(u) != Base.OneTo(length(u))` for
+# `u::TrixiMPIArray` in general, even if `u` and its underlying array use
+# one-based indexing.
+# Some consequences are that we need to implement specializations of `show`,
+# since the default ones call `length`. However, this doesn't work if not all
+# ranks call the same method, e.g., when showing an array only on one rank.
+Base.eachindex(u::TrixiMPIArray) = eachindex(parent(u))
+
+
+function Base.length(u::TrixiMPIArray)
+  local_length = length(parent(u))
+  return MPI.Allreduce(local_length, +, mpi_comm(u))
+end
+
+"""
+    local_length(u)
+
+Like `length(u)`, but returns the length of the local data for `u::TrixiMPIArray`.
+"""
+local_length(u) = length(u)
+local_length(u::TrixiMPIArray) = length(parent(u))
+
+
+"""
+    local_copy(u)
+
+Like `copy(u)`, but returns an unwrapped copy of the local data `parent(u)` for
+`u::TrixiMPIArray`.
```

Comment:
Why do we need this in the first place (just curious)? And wouldn't it suffice to just call `copy(parent(u))` where needed? Having an extra function seems overkill at the moment, at least if it is only used in one particular place for a particular reason.